### PR TITLE
[flutter_webrtc] Initial code

### PR DESCRIPTION
**Pull request by bwikbs**
Signed-off-by: MuHong Byun <mh.byun@samsung.com>

### CODE REVIEW

**Review by seungsoo47 — APPROVED** [review]
😊

**Comment by bbrto21** [comment]
[Guidelines](https://github.com/flutter-tizen/flutter-tizen/wiki/New-plugin-submission-guideline) for proposing new plugins have been published. Please read this.

I know you're working on the dev branch but too many files are being added at once. this makes reviews difficult. In particular, it seems that some of the original plugins are being copied as is, are these really necessary?

**Comment by bwikbs** [comment]
> Guidelines for proposing new plugins have been published. Please read this.

I've read it already.

> I know you're working on the dev branch but too many files are being added at once. this makes reviews difficult. In particular, it seems that some of the original plugins are being copied as is, are these really necessary?

As you mentioned, this is only dev branch and I think it is for convenient in many ways to work.
we've already developed this way in webview or google map and I don't understand why you're pointing this out now.
I hope we can only discuss the development direction, and other issues can be discussed when they are merged into master.
Isn't this still enough for you?

File: packages/flutter-webrtc/.gitignore [hunk 1]
```diff
@@ -0,0 +1,56 @@
+*.iml
```
Comment (swift-kim) [inline packages/flutter-webrtc/.gitignore:1]: The package name is `flutter_webrtc` (underscore between the words) but not `flutter-webrtc`. Rename the directory.

**Review by swift-kim — COMMENTED** [review]
As @bbrto21 pointed out, reviewability matters. If you're uploading code that only yourself can understand and provide no information about the code for reviewers, then there's no reason for you to make a PR until the code becomes reviewable.

**Comment by bwikbs** [comment]
> As @bbrto21 pointed out, reviewability matters. If you're uploading code that only yourself can understand and provide no information about the code for reviewers, then there's no reason for you to make a PR until the code becomes reviewable.

I still don't understand why we have to argue like this. ``Reviewability`` is also important to me.  
What I want to point out is that this is a dev branch and this is the initial code.(from somewhere)
I think it is a process of refining what is necessary and what is not. (Again, this is a dev branch.)

**Comment by bbrto21** [comment]
> we've already developed this way in webview or google map and I don't understand why you're pointing this out now.

I don't think it gives justification. however, I know this is a pretty convenient method.
Nevertheless, the gist of my question is "why do you have to copy so many files from the original except for examples?". 
In my understand we don't normally need this and the two plugins you mentioned are exceptional. If this plugin is also under an exceptional situation, I think it would be nice if you could share the reason with the reviewers.

I'm not saying this method is wrong. I hope some more information will be shared with reviewers and I just think that we should work together to provide a better development experience. that's all.

**Comment by bwikbs** [comment]
> Nevertheless, the gist of my question is "why do you have to copy so many files from the original except for examples?".
In my understand we don't normally need this and the two plugins you mentioned are exceptional. If this plugin is also under an exceptional situation, I think it would be nice if you could share the reason with the reviewers.

That's right. At this point, I don't think I need all these files either. This is a result of my laziness, and I'm not sure how far it's needed yet, so cleanup will happen once the first behavior is implemented.

> I'm not saying this method is wrong. I hope some more information will be shared with reviewers and I just think that we should work together to provide a better development experience. that's all.

I totally agree with you. I just wanted to discuss which direction to go form the original code (including the removal of unnecessary parts..)

**Comment by bbrto21** [comment]
> That's right. At this point, I don't think I need all these files either. This is a result of my laziness, and I'm not sure how far it's needed yet, so cleanup will happen once the first behavior is implemented.

If so, I would recommend merging the PRs after you clean up If there is no urgent reason.

If your only purpose is a quick proof-of-concept and code sharing, I don't think PR is necessary.
Also, for this purpose, I'm totally fine even if you or someone merge this.

I understand that my words may feel unfamiliar to you. It's the way we've been doing for a long time, but I'm suggesting a new perspective, so I hope you understand generously.

**Comment by bwikbs** [comment]
After some thinking about the current situation, I realized that the problem we are facing now is that we have a different way of thinking about running the dev branch.

> If so, I would recommend merging the PRs after you clean up If there is no urgent reason.
If your only purpose is a quick proof-of-concept and code sharing, I don't think PR is necessary.

Here's my opinion: (Just my opinion.)
I think that if you had a opinion on the dev branch, you freely post it as a PR, or tell me to delete these files as a specific comment. 
I think even if the purpose of this branch is only a quick proof-of-concept and code sharing, it still needs PR. Because it is also effective for that purpose.
I think this is going to be a collaboration of a few people and the dev branch was the canvas for that.

If you don't agree with this idea of the dev branch, I'll gladly drop the dev branch approach and prepare a PR for the master.
That PR will follow the [Guidelines](https://github.com/flutter-tizen/flutter-tizen/wiki/New-plugin-submission-guideline) and there will be no obsolete files.

**Comment by bbrto21** [comment]
> I think even if the purpose of this branch is only a quick proof-of-concept and code sharing, it still needs PR. Because it is also effective for that purpose. I think this is going to be a collaboration of a few people and the dev branch was the canvas for that.

I agree with most of what you said. there is no big difference between you and me. I think only difference between you and me is that I don't care what branch it is.
I just think PR should be convenient for others to review and including unnecessary parts should be suppressed as much as possible. It doesn't matter what branch it is. 

>> That's right. At this point, I don't think I need all these files either. This is a result of my laziness, and I'm not sure how far it's needed yet, so cleanup will happen once the first behavior is implemented.

> If so, I would recommend merging the PRs after you clean up If there is no urgent reason.

I still recommend above opinion, I would like to review your PR after it is clean enough.
and I'm still totally fine even if you or someone merge this.

> I think that if you had a opinion on the dev branch, you freely post it as a PR, or tell me to delete these files as a specific comment.

but if you don't want this, before I start review, I want to ask you politely again. 
Is there any reason for you to be in a hurry? if you don't, could you wait for me to review it?

**Comment by swift-kim** [comment]
> I just think PR should be convenient for others to review and including unnecessary parts should be suppressed as much as possible. It doesn't matter what branch it is.

I agree with this. If you want to discuss about the design of your plugin while the development is in progress, I think you can summarize your design decision and approaches as a comment in a [dedicated issue](https://github.com/flutter-tizen/plugins/issues/291), without requiring others to analyze and understand your WIP code.

**Comment by bwikbs** [comment]
> but if you don't want this, before I start review, I want to ask you politely again.
Is there any reason for you to be in a hurry? if you don't, could you wait for me to review it?

I also want to answer politely and clearly once again. 
Of course I can wait for your review. I've never been asked to do it fast. 😄 

> I agree with this. If you want to discuss about the design of your plugin while the development is in progress, I think you can summarize your design decision and approaches as a comment in a dedicated issue, without requiring others to analyze and understand your WIP code.

Now I think I fully understand what you mean. I also think what you are talking about is important.
I think further discussion is meaningless.

This PR(+flutter-webrtc-dev branch)  will close in a day or so. 
I will come back soon with a PR that everyone can be satisfied with.

**Comment by swift-kim** [comment]
@bwikbs Thank you for your kind understanding and I appreciate your hard work!

**Comment by bbrto21** [comment]
@bwikbs 

> Of course I can wait for your review. I've never been asked to do it fast. smile

Yes, of course you didn't say that. but I mean I felt that you are in urgent. because you didn't mention anything about my suggestion to clean up the PR. The reason I asked for a code cleanup before reviewing is because it takes a lot of time because there are so many things to review and I guess it takes more time to understand your work.

> If so, I would recommend merging the PRs after you clean up **If there is no urgent reason.**

I'm always grateful for your hard work and  I hope for your generous understanding.

**Comment by bwikbs** [comment]
@swift-kim @bbrto21 

I also appreciate you guys passion for your review and hard working!
Hopefully, we can discuss another topic next time. 😄

**Pull request closed without merging**